code/go/0chain.net/chaincore/state/state_debug_test.go

```diff
@@ -0,0 +1,171 @@
+package state
+
+import "testing"
+
+func TestDebug(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name       string
+		debugState int
+		want       bool
+	}{
+		{
+			name:       "TRUE",
+			debugState: DebugLevelChain,
+			want:       true,
+		},
+		{
+			name:       "FALSE",
+			debugState: DebugLevelNone,
+			want:       false,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			SetDebugLevel(tt.debugState)
```

Comment:
Take `SetDebugLevel` out of the t.Run block. `core\until.debugState` is only set once per execution, so is not protected against concurrent access. 

Probably best to think of `dedbugState` as a constant so only run parallel tests if they use the same setting for `debugState`.